Source/ProgressionSystemRuntime/Private/Data/PSSaveGameData.cpp

```diff
@@ -111,10 +138,30 @@ void UPSSaveGameData::NextLevelProgressionRowData()
 // Unlocks all levels and set maximum allowed progression points
 void UPSSaveGameData::UnlockAllLevels()
 {
+	TMap<FName, UPSSpotComponent*> SpotsMap = UPSWorldSubsystem::Get().GetAllSpotMap();
+
+	// unlock levels loop
 	for (TTuple<FName, FPSSaveToDiskData>& KeyValue : ProgressionSettingsRowDataInternal)
 	{
+		// levels
 		UnlockLevelByName(KeyValue.Key);
-		KeyValue.Value.CurrentLevelProgression = UPSWorldSubsystem::Get().GetCurrentProgressionSettingsRowByName().PointsToUnlock;
+		const FPSRowData& CurrentProgressionSettingsRowData = UPSWorldSubsystem::Get().GetRowDataByName(KeyValue.Key);
+		if (!ensureMsgf(CurrentProgressionSettingsRowData.IsValid(), TEXT("ASSERT: [%i] %hs:\n'CurrentProgressionSettingsRowData or Points to unlock = 0' is not valid!"), __LINE__, __FUNCTION__))
+		{
+			continue;
+		}
+		KeyValue.Value.CurrentLevelProgression = CurrentProgressionSettingsRowData.PointsToUnlock;
+
+		// skins
+		for (TTuple<FName, UPSSpotComponent*> Spot : SpotsMap)
```

Comment:
You dont need to iterate here manually to search for a key, just call `SpotsMap.Find(KeyValue.Key)` which returns pointer to a pointer (UPSSpotComponent**). If both pointers are valid (UPSSpotComponent != nullptr and *UPSSpotComponent != nullptr), then you found the match

Reply:
@moinm3uw

Reply:
acutally i did that initially but beacsue I was not able to find a way to access a pointer to a pointer  :) https://github.com/moinm3uw/ProgressionSystem/pull/59/commits/94927e7d26f018f9cd0d9e3c503f1ce8471c21d0